rpc/serve.go

```diff
@@ -0,0 +1,49 @@
+package rpc
+
+import (
+	"context"
+	"net"
+)
+
+// Serve serves a Cap'n Proto RPC to incoming connections
+// Serve exits with the listener error
+func Serve(lis net.Listener, opt *Options) error {
+	// Accept incoming connections
+	for {
+		rwc, err := lis.Accept()
+		if err != nil {
+			return err
+		}
+		// For each new incoming connection, create a new RPC transport connection that will serve incoming RPC requests
+		// rpc.Options will contain the bootstrap capability
+		go func() {
+			transport := NewStreamTransport(rwc)
+			conn := NewConn(transport, opt)
```

Comment:
One subtle thing here: you are going to want to make a copy of the options, cloning the bootstrap interface with `.AddRef()`. The reason for this is that the connection takes ownership of the bootstrap interface, so will drop it when the connection exits. So you want each connection to have its own reference. And you'll also want to drop the original reference when Serve() returns.

Reply:
@zenhack ,
Right, that is a good catch.  That said, Serve doesn't return until the connection is closed. Does this have to take into account that the bootstrap interface is used elsewhere or can be cancelled?

I wonder whether passing a bootstrap interface to the Serve function implies you hand over ownership. It is similar to obtaining a capability and handing it over elsewhere. The application is responsible to invoke AddRef if needed. Isn't that the same here?

Reply:
We should document it but, yeah that was my thinking: Serve should take ownership of the reference, for consistency with what NewConn does.

Reply:
> So you want each connection to have its own reference. And you'll also want to drop the original reference when Serve() returns.
Ohh I missed the point here. This looks like a major issue. So for each new incoming connection, use the Addref on Options.BootstrapClient and release when the  connection closes. Also release when Serve ends itself. Add note to documentation. Got it and will submit an update.

Reply:
Updated Serve with AddRef and update the testcase to better check for proper release